### PR TITLE
ci: [CNI] Add restart node in stage in the load test of cni pipeline

### PR DESCRIPTION
**Pull request by vipul-21**
**Reason for Change**:

- Added the test of restarting the nodes in the CNI load testing pipeline. It scales down the deployment and then restart the nodes without any delay.
- Update the `scale_deployment` script to include different parameters for CNI load testing

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests - No, tested in the pipeline CNI Load testing - https://msazure.visualstudio.com/One/_build/results?buildId=71712524&view=results

**Notes**:

### CODE REVIEW

**Review by camrynl — APPROVED** [review]

File: .pipelines/cni/cilium/cilium-cni-load-test.yaml [hunk 1]
```diff
@@ -56,8 +58,37 @@ stages:
...
+              azureSubscription: $(TEST_SUB_SERVICE_CONNECTION)
+              scriptLocation: "inlineScript"
+              scriptType: "bash"
+              addSpnToEnvironment: true
+              inlineScript: |
+                echo "Scale up the pods and immediated restart the nodes"
+                make -C ./hack/swift set-kubeconf AZCLI=az CLUSTER=${RESOURCE_GROUP}
+                make -C ./hack/swift azcfg AZCLI=az REGION=$(LOCATION)
+                echo "Scaling the pods down to 100 per node"
+                bash ./hack/scripts/scale_deployment.sh -n 0 -u 1000 -s
```
Comment (tamilmani1989) [inline .pipelines/cni/cilium/cilium-cni-load-test.yaml:78]: is it scaling down to 1000 or 100? also number of times is 0?

File: .pipelines/cni/cilium/cilium-cni-load-test.yaml [hunk 2]
```diff
@@ -56,8 +58,37 @@ stages:
...
+                echo "Restarting the nodes"
+                vmss_name=$(az vmss list -g MC_${RESOURCE_GROUP}_${RESOURCE_GROUP}_$(LOCATION) --query "[].name" -o tsv)
+                make -C ./hack/swift restart-vmss AZCLI=az CLUSTER=${RESOURCE_GROUP} REGION=$(LOCATION) VMSS_NAME=$vmss_name
+                bash ./hack/scripts/scale_deployment.sh -n 0 -u 1000 -c
```
Comment (tamilmani1989) [inline .pipelines/cni/cilium/cilium-cni-load-test.yaml:82]: why are we calling scale script again?

File: .pipelines/cni/cilium/cilium-cni-load-test.yaml (hunk 1 above)
Comment (vipul-21) [reply .pipelines/cni/cilium/cilium-cni-load-test.yaml:78]: The script scale up and down as a by default run. So by specifying `-n` as 0 we are telling it not to do that.
The flag `-s`  is used to define scale up operation. We are scaling down to 1000 pods (i.e 100 pods per node) from 240 pods per node.

File: .pipelines/cni/cilium/cilium-cni-load-test.yaml (hunk 2 above)
Comment (vipul-21) [reply .pipelines/cni/cilium/cilium-cni-load-test.yaml:82]: Similarly `-c` will wait until the deployment of 1000 pods is done. The same script is used.
I added the description of what the script can do. And checking the deployment status and deploying pods(scaling pods) can be done via passing flags to the script.
If we don't wait for the deployment to finish after the restart, the validate state script will fail as few of pods( sometimes) are still being created

**Pull request merged**